File: tests/rspec/pages/publish/_forum_tab.rb
class ForumTab < SitePrism::Section
  element :forum_title, 'input[name="forum__forum_title"]'
  element :forum_body, 'textarea[name="forum__forum_body"]'
  element :forum_id, 'div[data-input-value="forum__forum_id"]'
  elements :forum_id_choices, 'div[data-input-value="forum__forum_id"] .field-drop-choices label'
  element :forum_topic_id, 'input[name="forum__forum_topic_id"]'

  # Install forum, create a board, category, and forum
  def install_forum
    visit '/admin.php?/cp/addons'
    find('ul.toolbar a[data-post-url*="cp/addons/install/forum"]').click
    find('ul.toolbar a[href*="cp/addons/settings/forum"]').click

    # Create board
    find('.w-12 a[href*="cp/addons/settings/forum/create/board"]').click
    find('input[name="board_label"]').set 'Board'
    all('.w-12 button[value="save_and_close"]')[0].click

    # Create category
    find('.tbl-search a[href*="cp/addons/settings/forum/create/category/1"]').click
    find('input[name="forum_name"]').set 'Category'
    all('.w-12 button[value="save_and_close"]')[0].click

    # Create forum
    find('.tbl-action a[href*="cp/addons/settings/forum/create/forum/1"]').click
    find('input[name="forum_name"]').set 'Forum'
    all('.w-12 button[value="save_and_close"]')[0].click
  end
end

File: tests/cypress/cypress/elements/pages/publish/ForumTab.js
import ControlPanel from '../ControlPanel'

class ForumTab extends ControlPanel {
  constructor() {
    super()
    this.elements({
      'forum_title': 'input[name="forum__forum_title"]',
      'forum_body': 'textarea[name="forum__forum_body"]',
      'forum_id': 'div[data-input-value="forum__forum_id"]',
      'forum_id_choices': 'div[data-input-value="forum__forum_id"] .field-drop-choices label',
      'forum_topic_id': 'input[name="forum__forum_topic_id"]'
    })
  }

  // Install forum, create a board,
  install_forum() {
    cy.authVisit('/admin.php?/cp/addons')

    cy.get('input[title="select all"]').click()
      cy.get('select[name="bulk_action"]').select('Install')
      cy.get('button').contains('Submit').click()
       cy.get('input[type="submit"]').first().click()


    //cy.get('tbody > :nth-child(2) > :nth-child(1) > a').click() //GOTO Forums
    cy.visit('/admin.php?/cp/addons/settings/forum/create/board')


    cy.get('input[name="board_label"]').type('Board')
  
    cy.get('button[value="save_and_close"]').click()
   
    // // Create category
    // cy.get('td > .btn').click()//click create button
    // cy.get('.ajax-validate > .fieldset-required > .field-control > input').type('category')
    // cy.get('.form-btns-top > [value="save_and_close"]').click()

    // // //GOTO FORUM
    // cy.get('.no-results > .solo > a').eq(0).click()
    // // Create forum
    // cy.get('input[name="forum_name"]').type('Forum')
    
    // cy.get('.form-btns-top > [value="save_and_close"]').click()
    
  }

  create_entry(){
   
    cy.get('.t-0 > :nth-child(1) > .field-control > input').type('title')
   
    cy.get('a').contains('Forum').click()
    cy.get('.t-4 > :nth-child(1) > .field-control > input').type('title')
    cy.get('.field-control > textarea').type('body')
    cy.get('i').contains('Choose').click()
    cy.get('label').contains('board: Forum').click()
    
    
    cy.get('.form-btns-top > [value="save_and_close"]').click()
    cy.get('.app-notice__content > :nth-child(2)').contains("The entry")
    cy.get('.app-notice__content > :nth-child(2)').contains(" has been created")
  }

  get_post(){
    return cy.task('db:query', 'SELECT title, body FROM exp_forum_topics').then(function([rows,fields]){
      return rows;
    })
  }

 

   


}
export default ForumTab;




